components/homme/src/share/cxx/prim_driver.cpp

```diff
@@ -58,14 +57,16 @@ void initialize_dp3d_from_ps_c () {
   GPTLstop("tl-sc dp3d-from-ps");
 }
 
-void prim_run_subcycle_c (const Real& dt, int& nstep, int& nm1, int& n0, int& np1, const int& next_output_step)
+void prim_run_subcycle_c (const Real& dt, int& nstep, int& nm1, int& n0, int& np1, 
+                          const int& next_output_step, const int& nsplit_iteration)
 {
   GPTLstart("tl-sc prim_run_subcycle_c");
 
   auto& context = Context::singleton();
 
   // Get simulation params
   SimulationParams& params = context.get<SimulationParams>();
+  params.nsplit_iteration = nsplit_iteration;
```

Comment:
oh I did not realize it could be a copy. So, if prim_step_flexible has lines
```
  const auto& context = Context::singleton();
  const SimulationParams& params = context.get<SimulationParams>();
```
is it a copy of SP that prim_run_subcycle updated? Or is it a copy of some initial SP? _flexible routine depend on nsplit_iteration, but if it does not see it properly, then it is a big problem (though I am not sure how climo worked in this case).

Reply:
The method `context.get<T>()` returns a reference, so it is updating it. I was just pointing out that any place that stores a _copy_ of the params (like all functors), would not see the update.

Reply:
How about putting nsplit_iteration in TimeStep?

Reply:
You mean inside `TimeLevel`? Yeah, I think that's fine.

And tbc, as things are, I think it's fine also in SimulationParams. My (minor) concern is if someone reorganizes the code later on, and use nsplit_iteration from the params stored in a functor.

Reply:
Can I do it in next PR or do you want it here? just trying to avoid re-testing.

Reply:
Another PR is fine. Note also that Luca had previously suggested using `TimeLevel::nstep` and `SimulationParams::nsplit` to derive `nsplit_iteration` internally instead of passing it as an argument through the call stack: `nsplit_iteration = nstep % nsplit`. I think that's the best approach.

Reply:
@ambrad I will do as you guys suggest in the next PR, but I disagree with recomputing nsplit_iteration. It was already assigned somewhere, the most stable approach is to use it as input parameter in xx, not recompute it.

@mt5555 -- I have no opinion really, but EAM code in homme does not need nsplit_iteration because applycamforcing_tracers is done in dp_coupling layer. That itself is not perfect, if we remove it from dp layer, the easiest would be to use nsplit_iteration in prim_run_subcycle. Or, like you suggested before, to rewrite nsplit loop in dyn_comp. The latter is a lot of code changes, which can be done once simpler changes with tests are already in.

Reply:
the reason it's called from the dp layer in EAM is because it is doing a hard adjustment of the state output by the physics, and I think of this as a physics operation, not a dycore operation.   Thus, this code is more naturally part of the dp layer and not the dycore.  The cleanest example of this is ftype=1, which should be completely outside the dycore.   But if you really want it in the dycore, I dont understand why moving the nsplit loop from the dp layer to prim_driver is "a lot of code changes", it seems like it would just be ~10 lines.

as i said - this as an unsolicited opinion, feel free to ignore